test/ifElse/ifElseStep.js

```diff
@@ -368,4 +409,306 @@ describe('If Else step', () => {
+                                }
+                            }
+                        ]
+                    },
+                    elapsedTime: st.auditTrail.elapsedTime
+                })
+                assert.ok(st.auditTrail.elapsedTime > 0)
+            })
+        })
+
+        context('returning Ok from Else', () => {
+            const givenAnIfElseStep = () => {
+                const ifElseStep = ifElse({
+                    'If Step': step((ctx) => { return Ok(false) }),
+                    'Then Step': step({
+                        'nested step 1': step(() => { return Ok('response nestedstep 1') }),
+                        'nested step 2': step(() => { return Ok('response nestedstep 2') })
+                    }),
+                    'Else Step': step({
+                        'nested step 1': step(() => { return Err('response nestedstep 1') }),
+                        'nested step 2': step(() => { return Ok('response nestedstep 2') })
+                    })
+                })
+                return ifElseStep
+            }
+
+            it('should run', async () => {
+                //given
+                const st = givenAnIfElseStep()
+                //when
+                const ret = await st.run()
+                //then
+                assert.ok(ret.isErr)
+            })
+
+            it('should audit', async () => {
+                //given
+                const st = givenAnIfElseStep()
+                //when
+                await st.run()
+                //then
+                assert.deepStrictEqual(st.auditTrail, {
+                    type: 'if else',
+                    description: undefined,
+                    returnIf: {
+                        description: "If Step",
+                        type: "step",
+                        elapsedTime: st.auditTrail.returnIf.elapsedTime,
+                        return: { Ok: false },
+                    },
+                    returnElse: {
+                        description: "Else Step",
+                        type: "step",
+                        elapsedTime: st.auditTrail.returnElse.elapsedTime,
+                        return: {
+                            Error: "response nestedstep 1"
+                        },
+                        steps: [
+                            {
+                                description: "nested step 1",
+                                type: "step",
+                                elapsedTime: st.auditTrail.returnElse.steps[0].elapsedTime,
+                                return: {
+                                    Error: "response nestedstep 1"
+                                }
+                            }
+                        ]
+                    },
+                    elapsedTime: st.auditTrail.elapsedTime
+                })
+                assert.ok(st.auditTrail.elapsedTime > 0)
+            })
+        })
+    })
+
+    describe('using nested steps with context', () => {
```

Comment:
how `using nested steps with context` test is different from `using nested steps`?

Reply:
Nestedsteps use the _stopExecution which came from the context, if we overwrite the context with undefined, then the nestedstep will break.

Seeing this, I put an if and made two tests, with context and without context.

Reply:
Sorry, but I could not find the difference between these two test cases. It seems they are exercising the same code / scenario. Could you spot to me the difference?

Reply:
Right, in the ifElse step, we had `step.context = this.context` without if.

With this code in this manner, if we performed that code with usecase context, then will work correctly. But if we run the code without usecase context, the context of the step don't has the _stopExecution and will break the application...

I included an If statement to fix this , then this affects the two flows, for this reason I choose run the "same test" to both contexts.